src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7754,21 +7763,41 @@ protected override void VisitForEachExpression(BoundForEachStatement node)
             //       nested nullability of type parameters. See ForEach_22 for a concrete example of this.
             //    5. The collection type implements IEnumerable (non-generic). Because this version isn't generic, we don't need to
             //       do any reinference, and the existing conversion can stand as is.
-            //    6. The target framework's System.String doesn't implement IEnumerable. This is a compat case: System.String normally
+            //    6. The collection type implements the GetEnumerator pattern via an extension GetEnumerator. For this, there will be 
```

Comment:
I would expect this to be 7, not the backcompat String case. If it's not, then there _could_ theoretically be a compat break.

Reply:
I didn't think this was in order...